### PR TITLE
added form filler

### PR DESCRIPTION
**Pull request by mhilton**
Simple client for filling out form logins on the command line.

### CODE REVIEW

**Comment by mhilton** [comment]
Note: This feature was originally proposed in https://github.com/go-macaroon-bakery/macaroon-bakery/pull/67

File: form/form.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+//Package form enables the completion of forms specified as using environschema.
```
Comment (rogpeppe) [inline form/form.go]: // Package form provides ways to fill out forms corresponding to environschema
// schemas.

?

File: form/form.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	"sort"
+	"strings"
+
+	"github.com/juju/schema"
+	"golang.org/x/crypto/ssh/terminal"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/environschema.v1"
+)
+
+// PromptingFiller fills out a form described by an environscheama by
```
Comment (rogpeppe) [inline form/form.go]: // PromptingFiller fills out a set of environschema fields by
// prompting the user for each field in sequence.

?

File: form/form.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+// prompting for the response values.
+type PromptingFiller struct {
+	// The Prompter to use to get the responses if this is nil then
```
Comment (rogpeppe) [inline form/form.go]: s/responses if/responses. If

File: form/form.go [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+// Fill processes fields by first sorting them and then prompting for the
+// value of each one in turn.
+//
+// The fields are sorted by first by group name. Those in the same group
+// are sorted so that secret fields come after non-secret ones, finally
+// the fields are sorted by description.
+//
+// Each field will be prompted for, then the returned value will then be
```
Comment (rogpeppe) [inline form/form.go]: s/then be/be/

File: form/form.go [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+// validated against the type for the field. If the returned value does
```
Comment (rogpeppe) [inline form/form.go]: s/type for the field/field's type./
?

**Comment by rogpeppe** [comment]
Nice, LGTM thanks.

File: form/form.go (hunk 1 above)
Comment (mhilton) [inline form/form.go]: done

File: form/form.go (hunk 2 above)
Comment (mhilton) [inline form/form.go]: done

File: form/form.go (hunk 3 above)
Comment (mhilton) [inline form/form.go]: done

File: form/form.go (hunk 4 above)
Comment (mhilton) [inline form/form.go]: done

File: form/form.go (hunk 5 above)
Comment (mhilton) [inline form/form.go]: done

**Comment by mhilton** [comment]
To QA one can run formtest.

**Comment by urosj** [comment]
QA ok. Pin and password did not show while typing. It would be nice if the test would show ***\* while pin is entered, as we should have masked entry as well.

Otherwise :+1:

**Pull request merged**